backend/API/core/tests/tests.py

```diff
@@ -1,3 +1,415 @@
+        
+        #User email is incorrect
+        response = self.Perform_Test(self.invalid_email_payload)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+    def test_login_user_with_invalid_password(self):
+        '''Case to check if user login is possible where the password is invalid'''
+        
+        #User email exists but password is incorrect
+        response = self.Perform_Test(self.invalid_password_payload)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+
+class test_case_user_delete(APITestCase):
+    
+    @classmethod
+    def setUpTestData(self):
+        
+        u1 = {
+            "username": "test1",
+            "email": "test_user1@email.com",
+            "password": "bruhLOGIN1",
+            "password2": "bruhLOGIN1",
+            "date_of_birth": "2000-10-22T00:00:00Z"
+        }
+        u2 = {
+            "username": "test2",
+            "email": "test_user2@email.com",
+            "password": "bruhLOGIN2",
+            "password2": "bruhLOGIN2",
+            "date_of_birth": "2000-10-22T00:00:00Z"
+        }
+
+        
+        factory = APIRequestFactory()
+        request1 = factory.post('http://127.0.0.1:8000/api/register/', u1, format='json')
+        request2 = factory.post('http://127.0.0.1:8000/api/register/', u2, format='json')
+        user_view1 = RegistrationView.as_view()
+        user_view2 = RegistrationView.as_view()
+        response1 = user_view1(request1)
+        response2 = user_view2(request2)
+
+        if response1.status_code == status.HTTP_201_CREATED and response2.status_code == status.HTTP_201_CREATED:
+            pass
+        else:
+            raise Exception("Error occurred during creating a test user!")
+        
+        
+        self.user1_payload = {
+            "email": "test_user1@email.com",
+            "password": "bruhLOGIN1"
+        }
+        self.user2_payload = {
+            "email": "test_user2@email.com",
+            "password": "bruhLOGIN2"
+        }
+        self.user_invalid = {
+            "email": "iam_invalid@email.com",
+            "password": "iam_invalid"
+        }
+        
+        
+    def Perform_Test(self, user1, user2=None):
+        
+        #Logging into the user
+        factory_login = APIRequestFactory()
+        user_view_login = LoginView.as_view()
+        
+        request_login = factory_login.post('http://127.0.0.1:8000/api/login/',user1, format='json')
+        middleware = SessionMiddleware()
+        middleware.process_request(request_login)
+        request_login.session.save()
+        force_authenticate(request_login, user1)
+        response_login = user_view_login(request_login)
+    
+    
+        # header = {"Authorization":f"Bearer {response_login.data['access']}"}       
+        # factory_official.delete(f"{theURL}{x.id}", user2, format='json', **header, follow=True")
+    
+    
+        if response_login.status_code == status.HTTP_200_OK:
```

Comment:
if you can revert this if statement it would be more logical!

Reply:
Correct. Done